Makefile

```diff
@@ -9,6 +9,9 @@ venv:
 lint:
 	./.venv/bin/python -m flake8
 
+pylint:
```

Comment:
are both linters necessary? I am curious to see if flake8 can catch syntax "errors" that pylint cannot

Reply:
pylint is more rigorous than flake8, it runs slower but includes a lot more checks

Reply:
Ah so you mean what's the advantage of keeping flake8 now? I would say flake8 is great for quick check as I mentioned, as pylint runs slower. pylint also shows more errors in the repo. This PR addresses some of them but until we resolve all of them we cannot make pylint a blocking CI check. So it is good to keep flake8 until then atleast.